src/libraries/System.Text.Json/System.Text.Json.SourceGeneration.UnitTests/JsonSourceGeneratorTests.cs

```diff
@@ -177,46 +178,43 @@ public class ShouldNotFind { }
             string[] expectedFieldNamesMyType = { "PublicDouble", "PublicChar", "PrivateDouble", "PrivateChar" };
-            string[] receivedFieldNamesMyType = generator.foundTypes["MyType"].GetFields().Select(field => field.Name).ToArray();
-            Assert.Equal(expectedFieldNamesMyType, receivedFieldNamesMyType);
-
-            // Check for methods in created type.
-            string[] expectedMethodNamesMyType = { "get_PublicPropertyInt", "set_PublicPropertyInt", "get_PublicPropertyString", "set_PublicPropertyString", "get_PrivatePropertyInt", "set_PrivatePropertyInt", "get_PrivatePropertyString", "set_PrivatePropertyString", "MyMethod", "MySecondMethod" };
-            string[] receivedMethodNamesMyType = generator.foundTypes["MyType"].GetMethods().Select(method => method.Name).ToArray();
-            Assert.Equal(expectedMethodNamesMyType, receivedMethodNamesMyType);
+            string[] expectedPropertyNamesMyType = { "PublicPropertyInt", "PublicPropertyString", "PrivatePropertyInt", "PrivatePropertyString" };
+            string[] expectedMethodNamesMyType = { "get_PublicPropertyInt", "set_PublicPropertyInt", "get_PublicPropertyString", "set_PublicPropertyString", "get_PrivatePropertyInt", "set_PrivatePropertyInt", "get_PrivatePropertyString", "set_PrivatePropertyString", "MyMethod", "MySecondMethod", "UsePrivates" };
+            CheckFieldsPropertiesMethods("MyType", ref generator, expectedFieldNamesMyType, expectedPropertyNamesMyType, expectedMethodNamesMyType);
 
             // Check for NotMyType.
             Assert.Equal("System.Text.Json.Serialization.JsonConverterAttribute", generator.foundTypes["NotMyType"].FullName);
 
-            // Check for received properties in created type.
+            // Check for received fields, properties and methods for NotMyType.
+            string[] expectedFieldNamesNotMyType = { };
             string[] expectedPropertyNamesNotMyType = { "ConverterType" };
-            string[] receivedPropertyNamesNotMyType = generator.foundTypes["NotMyType"].GetProperties().Select(property => property.Name).ToArray();
-            Assert.Equal(expectedPropertyNamesNotMyType, receivedPropertyNamesNotMyType);
+            string[] expectedMethodNamesNotMyType = { "get_ConverterType", "CreateConverter" };
+            CheckFieldsPropertiesMethods("NotMyType", ref generator, expectedFieldNamesNotMyType, expectedPropertyNamesNotMyType, expectedMethodNamesNotMyType );
+        }
 
-            // Check for fields in created type.
-            string[] expectedFieldNamesNotMyType = { };
-            string[] receivedFieldNamesNotMyType = generator.foundTypes["NotMyType"].GetFields().Select(field => field.Name).ToArray();
-            Assert.Equal(expectedFieldNamesNotMyType, receivedFieldNamesNotMyType);
+        private void CheckFieldsPropertiesMethods(string typeName, ref JsonSerializerSourceGenerator generator, string[] expectedFields, string[] expectedProperties, string[] expectedMethods)
```

Comment:
Do the tests here depend on the order of the items returned by `.GetProperties/.GetFields` and co being deterministic? The order of the array of members returned by these reflection methods is not guaranteed, so the tests will eventually fail if dependent on deterministic ordering. This behavior is partly why we have feature requests like this in `JsonSerializer` - https://github.com/dotnet/runtime/issues/728.

Reply:
We now sort before comparison for non-dictionary based collections.